dask/_task_spec.py

```diff
@@ -101,18 +81,6 @@ def _identity_cast(*args, typ):
 _anom_count = itertools.count()
 
 
-@overload
-def parse_input(obj: _T_Iterable) -> _T_Iterable: ...
-
-
-@overload
-def parse_input(obj: TaskRef) -> Alias: ...
-
-
-@overload
-def parse_input(obj: _T_GraphNode) -> _T_GraphNode: ...
-
-
 def parse_input(obj: Any) -> object:
```

Comment:
This parse input can be used to wrap (recursively) user input or complex internal objects (e.g. in the array land we often have lists of lists that refer to other tasks). Instead of this function, I could also see a kwarg on the `Task` class that controls this behavior. Many options. This function is the simplest for now (I think) but I'm open for input